code/contracts/ONEWallet.sol

```diff
@@ -145,12 +151,13 @@ contract ONEWallet {
         _;
     }
 
+    // IH: why to not make just mapping of hashes to Commits? Then you might drop this and the next functions. Further, you won't have out-of-gas DOS by the attacker in commit stage => manifested here.
     function _findCommit(bytes32 hash) view internal returns (uint32, bool)
     {
         if (hash == "") {
             return (0, false);
         }
-        for (uint8 i = 0; i < commits.length; i++) {
+        for (uint8 i = 0; i < commits.length; i++) { // IH: uint8 might not be enough, I'd just go for uint since it is (cheap) in-memory var.
```

Comment:
Yes, should be uint32. The point on optimization is discussed above. Not sure whether in-memory operations are optimized at this point for shorter ints over longer ones (uint256), if not, I think Solidity should really fix it in future versions.

Reply:
addressed in https://github.com/polymorpher/one-wallet/commit/d289e3e3802c664ab73d97ec5aa79631f896af0f